lib/iris/cube.py

```diff
@@ -880,8 +880,11 @@ def convert_units(self, unit):
 
         """
         # If the cube has units convert the data.
-        if not self.units.is_unknown():
-            self.data = self.units.convert(self.data, unit)
+        if self.units.is_unknown():
+            raise iris.exceptions.UnitConversionError(
+                'cannot convert from unknown units to {!s}, '
+                'set the "units" attribute instead'.format(unit))
```

Comment:
>I think the second line of the error message may confuse rather than help

I disagree, a hint to the correct way to do this is highly valuable. The wording could be changed though. What did you think might confuse @dkillick?